Assets/_Main/Scripts/UI/PlayerHealthUIManager.cs

```diff
@@ -0,0 +1,146 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+
+/// <summary>
+/// --Health UI Manager--<para></para>
+/// 
+/// This script controls the UI for the health of the player
+/// Also has the upper limit for the number of hearts
+/// Its subcribed to the player events and updates the UI accordingly
+/// </summary>
+/// <remarks> Execution order should be after PlayerHealth. </remarks>
+/// 
+public class PlayerHealthUIManager : MonoBehaviour
+{
+    #region FIELDS
+
+    [Header("Player Data")]
+    [Tooltip("Singleton PlayerHealth script")]
+    [SerializeField] private PlayerHealth _playerHealth;
+
+    [Header("Hearts")]
+    [Tooltip("Gameobject prefab with the image of the full heart")]
+    [SerializeField] private GameObject _heart;
+    [SerializeField] private Image _heartImage;
+    [Tooltip("Gameobject prefab with the image of the hollow heart")]
+    [SerializeField] private GameObject _hollowHeart;
+    [SerializeField] private Image _hollowHeartImage;
+
+    private struct HeartUI // "Ivan" Como se nombran los structs
+    {
+        public GameObject _heartGO;
+        public Image _heartIMG;
+        public Animator _heartAnim;
+    }
+
+    private class HeartDictionary : Dictionary<int, HeartUI> // "Ivan" Como se nombran los diccionaries
+    {
+        private HeartUI newHeart;
+        internal void Add(int key, GameObject heartGO)
+        {
+            newHeart._heartGO = heartGO;
+            newHeart._heartIMG = heartGO.GetComponent<Image>();
+            newHeart._heartAnim = heartGO.GetComponent<Animator>();
+            this.Add(key, newHeart);
+        }
+    }
+
+    HeartDictionary _heartDictionary=new HeartDictionary();
+
+    #endregion
+
+    #region METHODS
+
+    /*
+    private void UpdateHeartUI()
+    {
+        // Podr�a revisar vida m�xima, vida actual y poner los corazones
+    }*/
```

Comment:
Normally we don't add this kind of things, what we do is add a ticket on Github as an idea for the future, maybe you need a new column for this kind of stuff.